src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
```

Comment:
This is the core of the change.  It runs the actual export on a background thread, displays a progress indicator,  blocks the user from interacting with the UI while the operation is running, and displays a success/failure message after the operation is complete.

There seems to be a lot going on across the two `saveScreenshot()` overloads that's unrelated to building the `ExportMenu`.  Therefore, I was tempted to extract a new class to isolate this functionality.  (Note also that the public `saveScreenshot()` method is called from outside this class by the `TripleAFrame` class.)  I held off because of my lack of knowledge of the TripleA architecture.

**Please advise if the public `saveScreenshot()` method should be extracted to a new class.**  If so, please suggest an appropriate package and class name.  I had previously considered `g.s.triplea.ui.export.ScreenshotExporter` based on what I saw related to the engine data exporters, but that may not be appropriate in this situation.